src/app/berkeley_migration_verifier/sql.ml

```diff
@@ -0,0 +1,332 @@
+
+  let get_all_blocks_till_height_query =
+    Caqti_request.find Caqti_type.int Caqti_type.int
+      {sql|
+          SELECT count(*) FROM blocks WHERE height <= ?;
+        |sql}
+
+  let get_all_blocks_till_height (module Conn : CONNECTION) height =
+    Conn.find get_all_blocks_till_height_query height
+
+  let get_account_accessed_count_query =
+    Caqti_request.find Caqti_type.unit Caqti_type.int
+      {sql| SELECT count(*) FROM accounts_accessed; |sql}
+
+  let count_account_accessed (module Conn : CONNECTION) =
+    Conn.find get_account_accessed_count_query ()
+
+  let get_account_id_accessed_in_commands_query =
+    Caqti_request.find Caqti_type.unit Caqti_type.int
+      {sql| 
+        select count(distinct ids.account_identifier_id) FROM 
+
+        ( 
+          select distinct account_identifier_id from accounts_accessed where account_identifier_id in 
+                ( select a.id from account_identifiers a inner join user_commands 
+                on public_key_id = fee_payer_id 
+                OR public_key_id = receiver_id 
+                OR public_key_id = source_id
+              )
+              
+              UNION ALL 
+        
+          select distinct account_identifier_id from accounts_accessed where account_identifier_id in 
+                      (  select a.id from account_identifiers a inner join internal_commands 
+                        on public_key_id = receiver_id
+                      )
+        ) as ids
+     
+      |sql}
+
+  let get_account_id_accessed_in_commands (module Conn : CONNECTION) =
+    Conn.find get_account_id_accessed_in_commands_query ()
+
+  module Block_info = struct
+    type t =
+      { id : int
+      ; parent_id : int
+      ; global_slot_since_genesis : int64
+      ; global_slot_since_hard_fork : int64
+      ; state_hash : string
+      ; height : int
+      ; chain_status : string
+      ; parent_hash : string
+      ; ledger_hash : string
+      ; protocol_version_id : int
+      }
+    [@@deriving hlist]
+
+    let typ =
+      Mina_caqti.Type_spec.custom_type ~to_hlist ~of_hlist
+        Caqti_type.
+          [ int; int; int64; int64; string; int; string; string; string; int ]
+
+    (* find all blocks, working back from block with given state hash *)
+    let forked_blockchain_query =
+      Caqti_request.collect Caqti_type.string typ
+        {sql| 
+                      SELECT id, 
+                             parent_id, 
+                             global_slot_since_genesis, 
+                             global_slot_since_hard_fork,
+                             state_hash,
+                             height,
+                             chain_status,
+                             parent_hash,
+                             ledger_hash,
+                             protocol_version_id
+                      
+                      FROM blocks
+                      WHERE state_hash = ? or protocol_version_id = 2 
```

Comment:
We should not relying on `id` that's auto-generated by sql. If you want version "2.0.0" then you should just search for this version and get its id.

Reply:
Removed explicit condition for protocol_version_id. Changed to get height bigger than fork point